src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php

```diff
@@ -47,6 +50,21 @@ public function collect(Request $request, Response $response, \Exception $except
     public function lateCollect()
     {
         $this->data['profile'] = serialize($this->profile);
+        $this->data['template_paths'] = array();
+
+        if (null !== $this->loader) {
+            $locator = (new \ReflectionObject($this->loader))->getMethod('findTemplate');
+            $locator->setAccessible(true);
```

Comment:
-1 for using private APIs of the Twig filesystem loader. This is calling for issues (there is no BC for internal APIs, even in patch releases).

Thus, your logic does not work in case multiple loaders are used (the template may have been loaded by a different loader and so come from a different file).

you should use public Twig APIs instead:

```php
$source = $twig->load($name)->getSourceContext();

$path = $source->getPath(); // Be careful. May be null if loader took it from elsewhere than the filesystem
```

Reply:
Maybe `$twig->loadTemplate($profile->getTemplate())->getSourceContext()->getPath()` instead?

Reply:
@ro0NL please, don't forget about this minor change needed to merge this. Thanks!